R/R/storedProcedureScripting.R

```diff
@@ -159,62 +205,74 @@ getRCode <- function(func, outputParams) {
     funcInputNames <- paste(funcInputNames, collapse = ", ")
 
     # add function call
+    #
     funcBody <- c(funcBody, paste0("result <- ", name,
                                    paste0("(", funcInputNames, ")")))
 
     # add appropriate ending
+    #
     ending <- getEnding(outputParams)
     funcBody <- c(funcBody, ending)
+
     return(paste0(funcBody, collapse = "\n"))
 }
 
 #
 # Get ending string
 # We change the result into an OutputDataSet - we only expect a single OutputDataSet result
-getEnding <- function(outputParams) {
+#
+getEnding <- function(outputParams)
+{
     outputDataSetName <- "OutputDataSet"
-    for(name in names(outputParams)) {
-        if (tolower(outputParams[[name]]) == "dataframe") {
+    for(name in names(outputParams))
+    {
+        if (tolower(outputParams[[name]]) == "dataframe")
+        {
             outputDataSetName <- name
         }
     }
+
     ending <- c( "if (is.data.frame(result)) {",
                  paste0("  ", outputDataSetName," <- result")
     )
 
-    if (length(outputParams) > 0) {
+    if (length(outputParams) > 0)
+    {
         ending <- c(ending, "} else if (is.list(result)) {")
 
-        for(name in names(outputParams)) {
-            if (tolower(outputParams[[name]]) == "dataframe") {
+        for(name in names(outputParams))
+        {
+            if (tolower(outputParams[[name]]) == "dataframe")
+            {
                 ending <- c(ending,paste0("  ", name," <- result$", name))
-            } else {
+            }
+            else
+            {
                 ending <- c(ending,paste0("  ", name, " <- result$", name))
             }
         }
+
         ending <- c(ending,
                     "} else if (!is.null(result)) {",
                     "  stop(\"the R function must return a list\")"
         )
     }
+
     ending <- c(ending, "}")
 }
 
-# @import RODBC
+# @import odbc
 # Execute the registration script
-register <- function(registrationScript, connectionString) {
+register <- function(registrationScript, connectionString)
+{
     output <- character(0)
 
-    tryCatch({
-        dbhandle <- odbcDriverConnect(connectionString)
-        output <- sqlQuery(dbhandle, registrationScript)
-    }, error = function(e) {
-        stop(paste0("Error in SQL Execution:\n", e))
-    }, finally ={
-        odbcCloseAll()
-    })
-    if (length(output) > 0 ) {
+    execute(connectionString, registrationScript)

```

Comment:
Is execute our function? Do we need try-catch here?

Reply:
executeInSQL.R line 208, the function has a try catch in itself